### ISSUE TITLE
Revisit decision to map "int" to LONG_TYPE in MyriaL

### ISSUE THREAD

**Issue by senderista** [uwescience/raco#461]
I understand this was due to overflow/precision concerns, but this is bloating the size of massive relations with many (small-valued) integer columns, like the SeaFlow data. Is it possible to auto-extend INT_TYPE to LONG_TYPE just in expressions with no schema specified?

**Comment by bmyerz** [comment]
can you give a myriaL example before and after the change?

**Comment by senderista** [comment]
Do you mean an example of query plans generated from the same MyriaL query before and after the change? This is just about mapping the `int` language keyword to raco types. Here is my motivating example:
```
O = load(
  "hdfs://vega.cs.washington.edu:8020/datasets/seaflow/opp_all.csv",
  csv(schema(
    cruise:string, file_time:string, particle:int,
    time:int, pulse_width:int, D1:int, D2:int, fsc_small:int, fsc_perp:int, fsc_big:int, pe:int, chl_small:int, chl_big:int))
);
store(O, armbrustlab:seaflow:opp_all, [cruise, file_time, particle]);
```
Right now all those `int` columns get mapped to `LONG_TYPE` (8-byte integer), even though they only have values on the order of 10^4-10^5. This is massively bloating the relations (and contributing to OOM errors that I'm dealing with right now). My suggestion is to just map them to `INT_TYPE` for storage and find a sensible heuristic for auto-extending to `LONG_TYPE` during computations that might overflow/lose precision. I'll admit to not having the context for the original decision though.

**Comment by bmyerz** [comment]
I don't have context for the original decision, either. My opinion at the moment is that Raco should support both LONG_TYPE and INT_TYPE and rely on the programmer to include `cast` `int` up to a `long` if they need to ensure 8-byte precision. 

We will probably have to change some `typeof` methods to implement useful implicit conversions. E.g. you ought to be able to call a `MyFunc(LONG_TYPE)` on an attribute of `INT_TYPE`.

**Comment by senderista** [comment]
I agree about supporting implicit conversions. Perhaps we could discuss this at the next Myria research meeting this Friday. @billhowe what do you think?
